File: lingting-spring-core/src/main/java/live/lingting/spring/event/SpringApplicationStartingListener.java
package live.lingting.spring.event;

import live.lingting.framework.context.ContextHolder;
import org.slf4j.Logger;
import org.springframework.boot.context.event.ApplicationStartingEvent;
import org.springframework.context.ApplicationListener;
import org.springframework.core.Ordered;
import org.springframework.core.annotation.Order;

/**
 * @author lingting 2023-12-06 17:18
 */
@Order(Ordered.HIGHEST_PRECEDENCE)
public class SpringApplicationStartingListener implements ApplicationListener<ApplicationStartingEvent> {

	private static final Logger log = org.slf4j.LoggerFactory.getLogger(SpringApplicationStartingListener.class);

	@Override
	public void onApplicationEvent(ApplicationStartingEvent event) {
		log.debug("spring application starting");
		ContextHolder.start();
	}

}

File: lingting-spring-core/src/main/kotlin/live/lingting/spring/event/SpringApplicationStartingListener.kt
package live.lingting.spring.event

import live.lingting.framework.context.ContextHolder.start
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.boot.context.event.ApplicationStartingEvent
import org.springframework.context.ApplicationListener
import org.springframework.core.Ordered
import org.springframework.core.annotation.Order

/**
 * @author lingting 2023-12-06 17:18
 */
@Order(Ordered.HIGHEST_PRECEDENCE)
class SpringApplicationStartingListener : ApplicationListener<ApplicationStartingEvent> {
    override fun onApplicationEvent(event: ApplicationStartingEvent) {
        log.debug("spring application starting")
        start()
    }

    companion object {
        private val log: Logger = LoggerFactory.getLogger(SpringApplicationStartingListener::class.java)
    }
}
